### PR TITLE
Add correlation loss as default for grid search

### PR DESCRIPTION
**Pull request by maltelueken**
Adds a `prfmodel.fitters.losses.CorrelationLoss` class that computes the Pearson correlation coefficient as a loss function.

This is the new default loss for `GridFitter` because it ignores differences in amplitude *and* baseline. Previously, the cosine similarity loss only ignored differences in amplitude but was sensitive to differences in baseline.

### CODE REVIEW

File: src/prfmodel/fitters/_grid.py [hunk 1]
```diff
@@ -49,17 +50,26 @@ class GridFitter:
     %(model_fitter)s
     %(stimulus)s
     loss : keras.optimizers.Loss or Callable, optional
-        Loss instance or function with the signature `f(y, y_pred)`, where `y` is the target data and `y_pred` are the
-        model predicitons. Default is `None` where a `keras.losses.CosineSimilarity` loss is used. Note that, when
-        a `keras.losses.Loss` instance is used, the argument `reduction` must be set to `"none"` to enable loss
-        computation for all data batches.
+        Loss instance or function with the signature `f(y_true, y_pred)`, where `y_true` is the target data and
+        `y_pred` are the model predicitons. Default is `None` where a
+        :class:`~prfmodel.fitters.losses.CorrelationLoss` loss is used. Note that, when a
```
Comment (angel-daza) [inline src/prfmodel/fitters/_grid.py:55]: If the loss passed here is just a `Callable` is there something to clarify here that emulates the `reduction="none"` param? Will the code still run fine? Does the callable always have to inherit from `keras.loss`?

**Review by angel-daza — APPROVED** [review]
All fine! just one minor clarification comment

File: src/prfmodel/fitters/_grid.py (hunk 1 above)
Comment (maltelueken) [reply src/prfmodel/fitters/_grid.py:55]: Thanks, I added a sentence on that.
